mlcp/src/main/java/com/marklogic/contentpump/LocalJobRunner.java

```diff
@@ -319,6 +317,32 @@ private int assignThreads(int splitIndex, int splitCount) {
             return threadCount / splitCount;
         }
     }
+
+    private void createThreadPool(Configuration conf) {
+        int numThreads;
+        if (threadCount != 0) {
+            // Use specified threadCount in the command line
+            numThreads = threadCount;
+        } else {
+            // Use server max thread count
+            numThreads =  conf.getInt(
+                MarkLogicConstants.SERVER_THREAD_COUNT, DEFAULT_THREAD_COUNT);
+            if (numThreads == 0) {
+                // ML server version is below 10.0-4.2, unable to get server
+                // thread count
+                numThreads = DEFAULT_THREAD_COUNT;
+            } else {
+                numThreads = (int)(0.5 * numThreads);
```

Comment:
The numTreads should be capped as we discussed (the client can never keep up with the max capacity of a very large cluster). And there could be multiple clients (they cannot all simultaneously use the max capacity). Can we make the cap as a parameter, otherwise we're asking for trouble :)  Ideally there should be smarter capacity/allocation negotiation between the client and server but we're not there yet.
For easier perf testing, does it make sensse to have the 0.5 as hidden parameters so it's easier for Perf team or us to test the best value. Thanks!

Reply:
Just try to stop me from setting it to `Integer.MAX VALUE`.